extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -242,48 +242,61 @@ export const HistoryItem = ({
               );
 
               if (!response.ok) {
-                throw new Error("failed to fetch token details");
-              }
-              const tokenDetails = await response.json();
+                const _err = await response.json();
+                captureException(
+                  `Failed to fetch token details - ${JSON.stringify(_err)}`,
+                );
 
-              const _token = {
-                contractId: attrs.contractId,
-                total: isRecieving ? attrs.amount : 0,
-                decimals: tokenDetails.decimals,
-                name: tokenDetails.name,
-                symbol: tokenDetails.symbol,
-              };
+                setRowText(operationString);
+                setTxDetails((_state) => ({
```

Comment:
This should have been falling back to a generic tx row in this edge case failure.